src/concretes/peer.ts

```diff
@@ -146,7 +126,7 @@ export class Peer implements IPeer {
                 // Add the new peer to our registry
                 const body = JSON.parse(message.body);
                 if (this.connectionService.GUID !== body.guid) {
-                    this.connectionService.routingTable.addPeer(body.guid, body.sender);
+                    this.connectionService.routingTable.addPeer(body.guid, body.guid, body.type);
```

Comment:
1 value is added twice as a parameter. Why not pass it once and use the same value twice in the .addPeer method?